subsys/net/lib/dns/llmnr_responder.c

```diff
@@ -134,29 +134,54 @@ static int bind_ctx(struct net_context *ctx,
 	return ret;
 }
 
-static void setup_dns_hdr(struct net_pkt *pkt, u16_t answers, u16_t dns_id)
+#define append(pkt, type, value)					\
+	do {								\
+		if (!net_pkt_append_##type##_timeout(pkt, value,	\
+						     BUF_ALLOC_TIMEOUT)) { \
+			ret = -ENOMEM;					\
+			goto drop;					\
```

Comment:
I tried to make the define to do "the right thing" meaning that all the functions would have error return at the end of the function. The defines are there so that the code looks a bit nicer and we do not have so many exposed error checks in the middle of the functions. I am not really sure I follow how would you like to see the code to look like in this case.

Reply:
I'm on @jukkar's side here, in the sense that I find such code acceptable and "attempting-to-do-better". It's not just a single case (like dns/llmnr_responder.c  to which @pfl originally commented), but a pattern reused in different files. It's effectively a poorman's exception handling handling, implemented in C.

Well, I can see @pfl's point too, that for someone not familiar with the code (or to someone who didn't look at the code for a couple of months ;-)), it may be confusing - after all, C is the language without exception handling, and it may be not immediately obvious that some functions/macros have a "side effect" of "throwing" an exception (goto'ing drop in this case). A compromise may be naming this macros e.g. as "append_or_drop()".